src/handler.ts

```diff
@@ -251,12 +259,12 @@ class FileHandler extends AbstractFileHandler {
    * Get the contents model for a widget.
    */
   protected getState(widget: Widget): Promise<IContentsModel> {
-    let path = AbstractFileHandler.pathProperty.get(widget);
-    let name = path.split('/').pop();
+    let model = AbstractFileHandler.modelProperty.get(widget);
+    let name = model.path.split('/').pop();
     name = name.split('.')[0];
     let content = (widget as CodeMirrorWidget).editor.getDoc().getValue();
-    return Promise.resolve({ path, content, name,
-                             type: 'file', format: 'text'});
+    return Promise.resolve({ path: model.path, content, name,
+                             type: model.type, format: model.format});
```

Comment:
So the path is ignored here, so is redundant (but needed?). The type should be 'file' and the format 'text', right? Here, we're handing back a specific kind of content that we absolutely know about.